src/main/kotlin/com/d3/chainadapter/adapter/ChainAdapter.kt

```diff
@@ -101,7 +107,7 @@ class ChainAdapter(
                     .subscribe({ block ->
                         publishUnreadLatch.await()
                         // Send only not read Iroha blocks
-                        if (block.blockV1.payload.height > lastReadBlock.get()) {
+                        if (block.blockV1.payload.height > lastReadBlock.get().toLong()) {
```

Comment:
Given the fact that `lastReadBlock.get()` returns `BigInteger`, we may go out of positive integers. I'd rather cast `block.blockV1.payload.heigh` instead.